myJoin.pl

```diff
@@ -36,7 +36,12 @@
 "help" => \$help) or die("Error in command line arguments -h for help\n");
 
 #process field argument
-@field = split(",", $field);
+@fields = split(",", $field);
+my %field_names = map {$_ => 1} @fields;
+
+if(@fields){
+	$header = 1;
+}
```

Comment:
This change is in line with our documentation. If the user supplies the `-f` flag, the `-H` flag is assumed. Here we codify that assumption.